iothub/device/src/Properties.cs

```diff
@@ -82,32 +89,48 @@ public IEnumerator<object> GetEnumerator()
             }
         }
 
+        /// <inheritdoc/>
         IEnumerator IEnumerable.GetEnumerator()
         {
             return GetEnumerator();
         }
 
         /// <summary>
-        /// Converts a <see cref="TwinProperties"/> collection to a properties collection
+        /// Gets the property from the collection.
+        /// </summary>
+        /// <remarks>
+        /// This calls <see cref="PayloadCollection.GetValue{T}(string)"/> and will use the serializer if needed. It is recommended to use this method over the <see cref="this[string]"/> accessor.
+        /// </remarks>
+        /// <typeparam name="T">The type to be returned.</typeparam>
+        /// <param name="propertyKey">The key of the property to be returned.</param>
+        /// <returns>A type of <typeparamref name="T"/> or <c>null</c></returns>
+        public T Get<T>(string propertyKey)
+        {
+            return _reportedPropertyCollection.GetValue<T>(propertyKey);
+        }
+
+        /// <summary>
+        /// Converts a <see cref="TwinProperties"/> collection to a properties collection.
         /// </summary>
-        /// <param name="twinProperties">The TwinProperties object to convert</param>
-        /// <returns></returns>
-        internal static Properties FromTwinProperties(TwinProperties twinProperties)
+        /// <param name="twinProperties">The TwinProperties object to convert.</param>
+        /// <param name="payloadConvention">A convention handler that defines the content encoding and serializer to use for the payload.</param>
+        /// <returns>A new instance of of the class from an existing <see cref="TwinProperties"/> using an optional <see cref="IPayloadConvention"/>.</returns>
+        internal static Properties FromTwinProperties(TwinProperties twinProperties, IPayloadConvention payloadConvention = default)
         {
             if (twinProperties == null)
             {
                 throw new ArgumentNullException(nameof(twinProperties));
             }
 
-            var writablePropertyCollection = new PropertyCollection();
+            var writablePropertyCollection = PropertyCollection.FromTwinCollection(twinProperties.Desired, payloadConvention);
```

Comment:
since we are creating the property collection from the .FromTwinCollection static helper, we don't need to iterate over the properties and add them in manually anymore